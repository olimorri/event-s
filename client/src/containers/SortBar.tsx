import React, { ChangeEvent } from 'react';
import {
  Flex,
  Box,
  FormControl,
  Accordion,
  AccordionItem,
  AccordionButton,
  AccordionPanel,
  Text,
  Checkbox,
  CheckboxGroup,
  Wrap,
} from '@chakra-ui/react';

const listOfTypes: string[] = [
  'Food',
  'Music',
  'Sport',
  'Leisure',
  'Outdoor',
  'Other',
];

export default function SortBar({
  checkboxes,
  setCheckboxes,
}: {
  checkboxes: string[];
  setCheckboxes: Function;
}) {
  let handleCheckboxChange = (e: ChangeEvent<HTMLInputElement>): void => {
    e.preventDefault();
    let selectedCheckboxes: string[] = [...checkboxes];
    if (!checkboxes.includes(e.target.value)) {
      selectedCheckboxes.push(e.target.value);
    } else {
      selectedCheckboxes.splice(selectedCheckboxes.indexOf(e.target.value), 1);
    }
    setCheckboxes(selectedCheckboxes);
  };

  return (
    <Box w={'100%'} justify={'center'} borderRadius="md" mb={10}>
      <FormControl justify={'center'}>
        <Flex justify={'center'} borderRadius="md">
          <Accordion w={'100%'} allowMultiple bg={'gray.100'} borderRadius="md">
            <AccordionItem w={'100%'}>
              <h2>
                <AccordionButton borderRadius="md">
                  <Box flex="1" textAlign="center">
                    Filter
                  </Box>
                </AccordionButton>
              </h2>
              <AccordionPanel pb={4}>
                <CheckboxGroup colorScheme="green">
                  <Text>Type:</Text>
                  <Wrap spacing="0px">
                    {listOfTypes.map((el) => (
                      <Checkbox
                        m={2}
                        onChange={handleCheckboxChange}
                        key={el}
                        value={el}
                      >
                        {el}
                      </Checkbox>
                    ))}
                  </Wrap>
                </CheckboxGroup>
              </AccordionPanel>
            </AccordionItem>
          </Accordion>
        </Flex>
      </FormControl>
    </Box>
  );
}

// {/* <HStack>
// <Checkbox onClick={handleCheckboxChange} value="ascending">Ascending</Checkbox>
// <Checkbox onClick={handleCheckboxChange} value="descending">Descending</Checkbox>
// <Checkbox onClick={handleCheckboxChange} value="Other">Other</Checkbox>
// </HStack> */}
