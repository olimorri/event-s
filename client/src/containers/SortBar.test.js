import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import SortBar from './SortBar';

const SortBarTest = () => (
  <BrowserRouter>
    <SortBar />
  </BrowserRouter>
);

describe('Snapshot', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<SortBarTest />);
    expect(asFragment(<SortBarTest />)).toMatchSnapshot();
  });
});

describe('Unit Testing', () => {
  it('should render the filter button', () => {
    render(<SortBarTest />);
    const attendBtn = screen.getByRole('button', {
      name: /filter/i,
    });
    expect(attendBtn).toBeInTheDocument();
  });
});
