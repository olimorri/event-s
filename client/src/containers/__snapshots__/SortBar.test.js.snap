// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-7b5fol"
    justify="center"
  >
    <div
      class="chakra-form-control css-1kxonj9"
      justify="center"
      role="group"
    >
      <div
        class="css-13jom7"
      >
        <div
          class="chakra-accordion css-1cojizo"
        >
          <div
            class="chakra-accordion__item css-1ncwuqs"
          >
            <h2>
              <button
                aria-controls="accordion-panel-1"
                aria-expanded="false"
                class="chakra-accordion__button css-evh648"
                id="accordion-button-1"
                type="button"
              >
                <div
                  class="css-1nwj029"
                >
                  Filter
                </div>
              </button>
            </h2>
            <div
              class="chakra-collapse"
              style="overflow: hidden; display: none; opacity: 0; height: 0px;"
            >
              <div
                aria-labelledby="accordion-button-1"
                class="chakra-accordion__panel css-b0qdn7"
                id="accordion-panel-1"
                role="region"
              >
                <p
                  class="chakra-text css-0"
                >
                  Type:
                </p>
                <div
                  class="chakra-wrap css-0"
                >
                  <ul
                    class="chakra-wrap__list css-1s0qoc1"
                  >
                    <label
                      class="chakra-checkbox css-1052anv"
                    >
                      <input
                        class="chakra-checkbox__input"
                        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                        type="checkbox"
                        value="Food"
                      />
                      <span
                        aria-hidden="true"
                        class="chakra-checkbox__control css-dnty2r"
                      />
                      <span
                        class="chakra-checkbox__label css-1oeb2oe"
                      >
                        Food
                      </span>
                    </label>
                    <label
                      class="chakra-checkbox css-1052anv"
                    >
                      <input
                        class="chakra-checkbox__input"
                        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                        type="checkbox"
                        value="Music"
                      />
                      <span
                        aria-hidden="true"
                        class="chakra-checkbox__control css-dnty2r"
                      />
                      <span
                        class="chakra-checkbox__label css-1oeb2oe"
                      >
                        Music
                      </span>
                    </label>
                    <label
                      class="chakra-checkbox css-1052anv"
                    >
                      <input
                        class="chakra-checkbox__input"
                        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                        type="checkbox"
                        value="Sport"
                      />
                      <span
                        aria-hidden="true"
                        class="chakra-checkbox__control css-dnty2r"
                      />
                      <span
                        class="chakra-checkbox__label css-1oeb2oe"
                      >
                        Sport
                      </span>
                    </label>
                    <label
                      class="chakra-checkbox css-1052anv"
                    >
                      <input
                        class="chakra-checkbox__input"
                        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                        type="checkbox"
                        value="Leisure"
                      />
                      <span
                        aria-hidden="true"
                        class="chakra-checkbox__control css-dnty2r"
                      />
                      <span
                        class="chakra-checkbox__label css-1oeb2oe"
                      >
                        Leisure
                      </span>
                    </label>
                    <label
                      class="chakra-checkbox css-1052anv"
                    >
                      <input
                        class="chakra-checkbox__input"
                        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                        type="checkbox"
                        value="Outdoor"
                      />
                      <span
                        aria-hidden="true"
                        class="chakra-checkbox__control css-dnty2r"
                      />
                      <span
                        class="chakra-checkbox__label css-1oeb2oe"
                      >
                        Outdoor
                      </span>
                    </label>
                    <label
                      class="chakra-checkbox css-1052anv"
                    >
                      <input
                        class="chakra-checkbox__input"
                        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                        type="checkbox"
                        value="Other"
                      />
                      <span
                        aria-hidden="true"
                        class="chakra-checkbox__control css-dnty2r"
                      />
                      <span
                        class="chakra-checkbox__label css-1oeb2oe"
                      >
                        Other
                      </span>
                    </label>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
